src/python/pants/backend/java/compile/javac.py

```diff
@@ -199,16 +221,31 @@ async def compile_java_source(
         # a `package-info.java` in a single partition.
         jar_output_digest = EMPTY_DIGEST
 
+    output_classpath = ClasspathEntry(
+        jar_output_digest, (output_file,), direct_dependency_classpath_entries
+    )
+
+    if export_classpath_entries:
+        merged_export_digest = await Get(
+            Digest,
+            MergeDigests((output_classpath.digest, *[i.digest for i in export_classpath_entries])),
```

Comment:
```suggestion
            MergeDigests((output_classpath.digest, *(i.digest for i in export_classpath_entries))),
```
To avoid creating an intermediate list.